Source/Core/Common/FileSearch.cpp

```diff
@@ -45,57 +45,17 @@ FileSearchWithTest(const std::vector<std::string>& directories, bool recursive,
   return result;
 }
 
-static std::vector<std::string> DoFileSearchNoSTL(const std::vector<std::string>& directories,
-                                                  const std::vector<std::string>& exts,
-                                                  bool recursive)
-{
-  bool accept_all = exts.empty();
-  return FileSearchWithTest(directories, recursive, [&](const File::FSTEntry& entry) {
-    if (accept_all)
-      return true;
-    std::string name = entry.virtualName;
-    std::transform(name.begin(), name.end(), name.begin(), ::tolower);
-    return std::any_of(exts.begin(), exts.end(), [&](const std::string& ext) {
-      return name.length() >= ext.length() &&
-             name.compare(name.length() - ext.length(), ext.length(), ext) == 0;
-    });
-  });
-}
-
-std::vector<std::string> DoFileSearch(const std::vector<std::string>& directories,
-                                      const std::vector<std::string>& exts, bool recursive)
-{
-  return DoFileSearchNoSTL(directories, exts, recursive);
-}
-
 #else
 
-std::vector<std::string> DoFileSearch(const std::vector<std::string>& directories,
-                                      const std::vector<std::string>& exts, bool recursive)
+std::vector<std::string>
+DoFileSearch(const std::vector<std::string>& directories, bool recursive,
+             std::function<bool(const std::string& path, bool is_directory)> filter)
 {
-  bool accept_all = exts.empty();
-
-  std::vector<fs::path> native_exts;
-  for (const auto& ext : exts)
-    native_exts.push_back(ext);
-
-  // N.B. This avoids doing any copies
-  auto ext_matches = [&native_exts](const fs::path& path) {
-    const auto& native_path = path.native();
-    return std::any_of(native_exts.cbegin(), native_exts.cend(), [&native_path](const auto& ext) {
-      // TODO provide cross-platform compat for the comparison function, once more platforms
-      // support std::filesystem
-      return native_path.length() >= ext.native().length() &&
-             _wcsicmp(&native_path.c_str()[native_path.length() - ext.native().length()],
-                      ext.c_str()) == 0;
-    });
-  };
-
   std::vector<std::string> result;
   auto add_filtered = [&](const fs::directory_entry& entry) {
-    auto& path = entry.path();
-    if (accept_all || (ext_matches(path) && !fs::is_directory(path)))
-      result.emplace_back(path.u8string());
+    const std::string u8_path = entry.path().u8string();
```

Comment:
This also does a copy, but we more or less need this for this PR's design to work... So I suppose this PR's design is unworkable if we care about performance.